src/App.tsx

```diff
@@ -1,24 +1,148 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { FilterOption } from './types/FilterOption';
+import { getTodos, createTodo, deleteTodo } from './api/todos';
+import { AddTodoInput } from './components/AddTodoInput';
+import { Footer } from './components/Footer';
+import { Alert } from './components/Alert';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 10527;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(FilterOption.All);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
```

Comment:
please add type to `errorMessage`. and there is no need to 2 state variables. `hasError` can be calculated based on `errorMessage`
```suggestion
  const [errorMessage, setErrorMessage] = useState('');
  const hasError = errorMessage !== '';
```

Reply:
That's a great suggestion, thank you! -Done